app/filtering.js

```diff
@@ -520,14 +521,29 @@ function registerForDownloadListener (session) {
     }
 
     const defaultPath = path.join(getSetting(settings.DOWNLOAD_DEFAULT_PATH) || getSetting(settings.DEFAULT_DOWNLOAD_SAVE_PATH) || app.getPath('downloads'), itemFilename)
-    const savePath = ((process.env.SPECTRON || (!getSetting(settings.DOWNLOAD_ALWAYS_ASK) && !item.promptForSaveLocation())) ? defaultPath : dialog.showSaveDialog(win, { defaultPath }))
+    let savePath = ((process.env.SPECTRON || (!getSetting(settings.DOWNLOAD_ALWAYS_ASK) && !item.promptForSaveLocation())) ? defaultPath : dialog.showSaveDialog(win, { defaultPath }))
```

Comment:
I think there should be a single `if` statement here that wraps all of this, but can you switch `!getSetting(settings.DOWNLOAD_ALWAYS_ASK) && !item.promptForSaveLocation()` to `getSetting(settings.DOWNLOAD_ALWAYS_ASK)  || item.promptForSaveLocation()`? The former is confusing